lightning/src/routing/gossip.rs

```diff
@@ -1605,7 +1690,7 @@ impl<L: Deref> NetworkGraph<L> where L::Target: Logger {
 			let mut nodes = self.nodes.write().unwrap();
 			for scid in scids_to_remove {
 				let info = channels.remove(&scid).expect("We just accessed this scid, it should be present");
-				Self::remove_channel_in_nodes(&mut nodes, &info, scid);
+				self.remove_channel_in_nodes(&mut nodes, &info, scid);
```

Comment:
So in this case we still want to remove the channels but just not track them? If that's the case should we have a bool param that indicates if we want to track or not (redundant for no-std of course)

Reply:
Well I guess Matt ultimately wants to handle `channel_update` fine even for these nodes, according to `update_channel_intern`.

The part which might not work anymore is
`			for chan_id in node.channels.iter() {
				let chan = channels.get(chan_id).unwrap();
`

Worth testing at least?

Reply:
What could work i guess is passing a bool to `remove_channel_in_nodes` specifying whether you want `remove_from_node!(chan.node_one);` or you're fine just doing everything else. But I might be wrong :)

Reply:
Yeah! That's kind of my thinking, although not sure how awkward it would be. I'll whip something up with a test case.

Reply:
This is still not resolved, right?

Reply:
Ah, it was resolved in (maybe badly named) https://github.com/lightningdevkit/rust-lightning/pull/1649/commits/31d596a129c1a56a683eb6a49b208dfcab96473a according to Matt's comment. We move the insertion into `channel_failed` instead and make sure not to insert into `removed_nodes` just because its last channel was removed (although it does still get removed from the graph until resync)

Reply:
Ahhh sounds good. Some ideas for the tests:
- call `remove_stale_channels_with_time` or `remove_stale_channels_and_tracking`
- try to re-announce the channel 
- see that it is indeed stored (not ignored due to `removed_channels`)

Not a big deal, because I'll have to flip that test later in #1671 anyway

A similar test would work for `removed_nodes` and nodes re-adding after the same calls i guess.

Although these tests seem kinda artificial.